core/audio.go

```diff
@@ -63,45 +82,59 @@ func (a *AudioSystem) Remove(basic ecs.BasicEntity) {
 	}
 }
 
-func (a *AudioSystem) New(*ecs.World) {
+func (a *AudioSystem) New(w *ecs.World) {
 	a.cachedVolume = MasterVolume
 
 	if a.HeightModifier == 0 {
 		a.HeightModifier = defaultHeightModifier
 	}
 
-	if err := al.OpenDevice(); err != nil {
-		log.Println("Error initializing AudioSystem:", err)
+	if !audioSystemPreloaded {
+		AudioSystemPreload()
+	}
+
+	var cam *cameraSystem
+	for _, system := range w.Systems() {
+		switch sys := system.(type) {
+		case *cameraSystem:
+			cam = sys
+		}
+	}
+
+	if cam == nil {
+		log.Println("[ERROR] CameraSystem not found - have you added the `RenderSystem` before the `AudioSystem`?")
```

Comment:
Hmm - but if they *did* add it before the `AudioSystem`, then it might confuse even more?